pulp_container/app/registry_api.py

```diff
@@ -1110,9 +1111,11 @@ def get_content_units_to_add(self, manifest, tag):
         return add_content_units
 
     def fetch_manifest(self, remote, pk):
-        relative_url = "/v2/{name}/manifests/{pk}".format(
-            name=remote.namespaced_upstream_name, pk=pk
-        )
+        try:
+            repo_name = filter_repo(remote)
+        except RepositoryNotFound:
+            raise
```

Comment:
> Do you think it makes more sense to move the check to a different place?

Oh!! Nice!
At first, I thought that putting it there would be good because we would be *failing early* (before downloading the manifest), but I didn't think about the previous resources that had been created.